CHANGELOG.md

```diff
@@ -51,6 +51,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Added class name prefix to metrics logged by `DeviceStatsMonitor` ([#12228](https://github.com/PyTorchLightning/pytorch-lightning/pull/12228))
 
 
+- Automatically wrap custom samplers under a distributed environment by using Catalyst's `DistributedSamplerWrapper` ([#12959](https://github.com/PyTorchLightning/pytorch-lightning/pull/12959))
```

Comment:
Are we setting as a dependency or what is the relation?

Reply:
No, we are not. We adapted two classes from their codebase into ours.